wcfsetup/install/files/lib/acp/page/IndexPage.class.php

```diff
@@ -170,6 +170,15 @@ public function assignVariables()
             }
         }
 
+        $taintedApplications = [];
+        foreach (ApplicationHandler::getInstance()->getApplications() as $application) {
```

Comment:
Maybe use `array_filter()` instead? Although its syntax is a bit awkward, it does improve readability quite a lot. Also, do we even care for the `packageID` as the index?

Reply:
I very much like `array_filter`, but it's also like this for consistency with the code above. The `packageID` in the index is used for consistency with the PackageList, which needs it to show the warning icon besides the package name.